### PR TITLE
Better handle periodicals in response

### PR DESCRIPTION
**Pull request by mrdpopov**
## Purpose

Better handle periodicals in response

## Approach

Using https://github.com/folio-org/mod-rtac/pull/46 as a baseline. Adding support for a new parameter "fullPeriodicals". 

*    if it’s set to “true”, then item-level information is added to all periodicals without holdings-level information,
*    if it’s set to “false” or not defined, then holding-level information replaces item-level for periodicals.

Jira: https://issues.folio.org/browse/MODRTAC-37
Jira: https://issues.folio.org/browse/MODRTAC-38

### CODE REVIEW

**Comment by craigmcnally** [comment]
It would be helpful if we could move towards smaller PRs.  This PR encompasses 2 JIRAs and thousands of line changes making it time consuming to review.

**Comment by mrdpopov** [comment]
Thank you for the feedback. This pull request is based on the major refactoring in another pull request #46, hence they can't go separately. Once the latter is merged, only a small bunch of changes related to better periodical handling will remain here.

**Comment by craigmcnally** [comment]
> Thank you for the feedback. This pull request is based on the major refactoring in another pull request #46, hence they can't go separately. Once the latter is merged, only a small bunch of changes related to better periodical handling will remain here.

Ah, ok I missed that part.  Thanks for clarifying

File: src/main/java/org/folio/mappers/FolioToRtacMapper.java
Comment (craigmcnally) [inline src/main/java/org/folio/mappers/FolioToRtacMapper.java:26]: Is there a better way to handle this besides hardcoding these values here?

Comment (craigmcnally) [reply src/main/java/org/folio/mappers/FolioToRtacMapper.java:26]: isn't nature-of-content a controlled vocabulary?  So what happens if a tenant replaces these values, or wants to use different values?  Maybe they remove these and just want a single entry "periodical", or add "magazine", etc.?

File: src/main/java/org/folio/mappers/FolioToRtacMapper.java
Comment (craigmcnally) [inline src/main/java/org/folio/mappers/FolioToRtacMapper.java:133]: again, isn't mode-of-issuance a controlled vocabulary?  Hardcoding "serial" here makes this fragile?

Comment (craigmcnally) [reply src/main/java/org/folio/mappers/FolioToRtacMapper.java:133]: You can't assume the reference data will always be present.  Some (actually most) institutions have chosen to not use the OOTB FOLIO reference data

File: src/main/java/org/folio/mappers/FolioToRtacMapper.java
Comment (mrdpopov) [reply src/main/java/org/folio/mappers/FolioToRtacMapper.java:26]: Thanks for pointing this out. I had no information that was somewhere in tenant settings

File: src/main/java/org/folio/mappers/FolioToRtacMapper.java
Comment (mrdpopov) [reply src/main/java/org/folio/mappers/FolioToRtacMapper.java:134]: As it was stated in the requirements in MODRTAC-37, an instance is considered to be a  periodical when either:
* mode of issuance"= serial
* or  "nature of content" is one of (journal, newspaper).

I think, even if theese values are absent in a tenant, there will be no error, the system just would not be able to tell if an instance is a periodical or not.

Can you suggest a solution here?

File: src/main/java/org/folio/mappers/FolioToRtacMapper.java
Comment (mrdpopov) [reply src/main/java/org/folio/mappers/FolioToRtacMapper.java:26]: As it was stated in the requirements in MODRTAC-37, an instance is considered to be a periodical when either:

    mode of issuance"= serial
    or "nature of content" is one of (journal, newspaper).

I think, even if theese values are absent in a tenant, there will be no error, the system just would not be able to tell if an instance is a periodical or not.

**Review by craigmcnally — APPROVED** [review]
Approving with the stipulation that test coverage is brought up to 80%+.  My understanding is that you're currently working on this.

**Comment by craigmcnally** [comment]
In a separate conversation it was discussed that at at minimum the expectation that these modes of issuance, nature of content values are used is documented in the release notes, as well as somewhere highly visible to implementers, e.g. API docs via RAML?

Additionally, a story should be created capturing the technical debt associated with hardcoding these values.  Other teams have solved similar problems by having the module inspect configuration which informs the it which values to use.

**Pull request merged**
